.github/workflows/delete-branch-on-pr-close.yaml

```diff
@@ -0,0 +1,22 @@
+---
```

Comment:
I'm not sure whether we should delete the branch in such an aggressive way. For example, @phanikumv wants to keep the https://github.com/astronomer/astronomer-providers/tree/add-postgres-async, but with this action, we'll remove that as well. I'm thinking of doing it only for the testing rc branches

Reply:
yes we need to cleanup for rc test branches only

Reply:
we can always restore the branch if we accidentally delete the branch or upon PR closure. I don't think deletion is permanent delete. But will check that we've such a prefix.

Reply:
This action will only delete the branch when a PR is closed. And branches deleted from close PRs can be restored AFAIK https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-branches-in-your-repository/deleting-and-restoring-branches-in-a-pull-request#restoring-a-deleted-branch

Reply:
Wow, I didn't know it. Thanks for sharing this!